PyPDF2/pdf.py

```diff
@@ -1992,41 +1992,6 @@ def _pairs(self, array):
             if (i+1) >= len(array):
                 break
 
-    def readNextEndLine(self, stream):
-        debug = False
-        if debug: print(">>readNextEndLine")
-        line = b_("")
-        while True:
-            # Prevent infinite loops in malformed PDFs
-            if stream.tell() == 0:
-                raise utils.PdfReadError("Could not read malformed PDF file")
-            x = stream.read(1)
-            if debug: print(("  x:", x, "%x"%ord(x)))
-            if stream.tell() < 2:
-                raise utils.PdfReadError("EOL marker not found")
-            stream.seek(-2, 1)
-            if x == b_('\n') or x == b_('\r'): ## \n = LF; \r = CR
-                crlf = False
-                while x == b_('\n') or x == b_('\r'):
-                    if debug:
-                        if ord(x) == 0x0D: print("  x is CR 0D")
-                        elif ord(x) == 0x0A: print("  x is LF 0A")
-                    x = stream.read(1)
-                    if x == b_('\n') or x == b_('\r'): # account for CR+LF
-                        stream.seek(-1, 1)
-                        crlf = True
-                    if stream.tell() < 2:
```

Comment:
There are also some weird quirks here - some might be intentional (e.g. this skips empty lines, which behavior I have preserved), but others are probably not, e.g. this boundary check means that the line before the previous one matters, and this function can also skip past one-character lines when `crlf == True`:

```
r = PdfFileReader("whatever.pdf")
x = io.BytesIO(b'foo\nbar\nX\r\nbaz')
x.seek(-1, 2)
r.readNextEndLine(x) # "baz"
r.readNextEndLine(x) # "bar" - "X" has been skipped!
```